app/src/main/kotlin/com/hippo/ehviewer/ui/settings/DownloadScreen.kt

```diff
@@ -157,6 +159,26 @@ fun DownloadScreen(navigator: DestinationsNavigator) {
                 summary = stringResource(id = R.string.settings_download_archive_metadata_summary),
                 value = Settings::archiveMetadata,
             )
+            WorkPreference(
+                title = stringResource(id = R.string.settings_download_reload_metadata),
+                summary = stringResource(id = R.string.settings_download_reload_metadata_summary),
+            ) {
+                DownloadManager.downloadInfoList.filter { it.state == DownloadInfo.STATE_FINISH }.apply {
+                    runSuspendCatching {
+                        fillGalleryListByApi(this, EhUrl.referer)
+                        parMapNotNull {
+                            if (it.pages != 0) {
```

Comment:
`val toUpdate = filter {  it.pages != 0 }.parMap { di -> di.galleryInfo.also { SpiderDen(it, di.dirname!!).writeComicInfo(false) } }
EhDB.updateGalleryInfo(toUpdate)`

Reply:
It's not necessary to filter as the API is not likely to give 0 on pages.
But I suggest skipping those items that already have metadata to reduce network requests.

Reply:
It appears that the tags for some galleries are altered post-upload, resulting in a discrepancy between my metadata/DB and E-Hentai.org .
For instance, some galleries are labeled as 'Unknown' in `DownloadsScreen`(from DB query), but show an artists in GalleryDetailScreen(from EhEngine).
Regrettably, I tests code using my storage files which resulted in the original files being overwritten, hence I am unable to provide video evidence to substantiate this.

Reply:
Yeah I'm aware of that.
Let's consider galleries posted more than one month ago "stable" and their tags won't change.